pr_agent/algo/ai_handler.py

```diff
@@ -99,28 +109,39 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
                 )
             if self.azure:
                 model = 'azure/' + model
+            system = get_settings().get("CONFIG.MODEL_SYSTEM_PREFIX", "") + system + \
+                     get_settings().get("CONFIG.MODEL_SYSTEM_SUFFIX", "")
+            user = get_settings().get("CONFIG.MODEL_USER_PREFIX", "") + user + \
+                   get_settings().get("CONFIG.MODEL_USER_SUFFIX", "")
             messages = [{"role": "system", "content": system}, {"role": "user", "content": user}]
             response = await acompletion(
                 model=model,
                 deployment_id=deployment_id,
                 messages=messages,
                 temperature=temperature,
-                force_timeout=get_settings().config.ai_timeout
+                force_timeout=get_settings().config.ai_timeout,
+                api_base=self.api_base,
             )
```

Comment:
**Suggestion:** In the `chat_completion` method, the `force_timeout` parameter is set to `get_settings().config.ai_timeout`. This means that the timeout value is constant for all the API calls. It might be beneficial to implement a dynamic timeout strategy, where the timeout value increases with each retry. This can be achieved by using an exponential backoff strategy for the timeout value. [enhancement]